_includes/search-lunr.html

```diff
@@ -0,0 +1,71 @@
+<script src="/js/lunr.js"></script>
+
+<script>
+{% assign counter = 0 %}
+var documents = [{% for page in site.pages %}{% if page.url contains '.xml' or page.url contains 'assets' or page.search == 'exclude' %}{% else %}{
```

Comment:
Since the index is client-side, this will put all the pages with all their content into this JS snippet. This will become a pretty big chunk of text for larger blogs pretty quickly. 

It should work great as long as the search lives on a separate page. If we'd want to have a search bar in the header that searches the blog, on any page — that would mean that we'll download the full blog on every page.

I don't know if there's a good workaround for this — it's definitely good enough for now, until we see any problems and issues from our users. We could go ahead and use Algolia instead, but that's a 3rd party dependency I'd rather not have, or have as an opt-in.

Reply:
That's good to know, I guess we will have search on a separate page for now